framework/src/application.ts

```diff
@@ -377,13 +370,8 @@ export class Application {
 	// Private
 	// --------------------------------------
 
-	private _registerModule(mod: BaseModule, validateModuleID = false): void {
+	private _registerModule(mod: BaseModule): void {
 		assert(mod, 'Module implementation is required');
-		if (validateModuleID && mod.id.readInt32BE(0) < MINIMUM_EXTERNAL_MODULE_ID) {
```

Comment:
should we validate that name string is unique also?

Reply:
https://github.com/LiskHQ/lisk-sdk/pull/7415/files/82465d035550e69f6993f539cbd562736537db4d#diff-cb2314f09fb36abaa9b56f7e516e6753a48441c7e8ac30478fd6ad2a9f998360R295

validation is checked here